src/utils/formatting.py

```
@@ -59,27 +59,60 @@ def utc_now_iso() -> str:
         datetime.datetime(2023, 1, 1, 12, 0, tzinfo=datetime.timezone.utc)
-        >>> parse_datetime_iso('2023-01-01T12:00:00+00:00')
-        datetime.datetime(2023, 1, 1, 12, 0, tzinfo=datetime.timezone.utc)
+        >>> parse_datetime_iso('2023-01-01T12:00:00+05:00')
+        datetime.datetime(2023, 1, 1, 12, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=18000)))
     """
-    # Convert Z format to +00:00 format for Python's fromisoformat
-    normalized_string = iso_string.replace("Z", "+00:00")
-    return datetime.fromisoformat(normalized_string)
+    # Input validation - ensure type and reject suspicious content
+    if not iso_string:
+        raise ValueError("Invalid input: must be a non-empty string")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)